app/models/manageiq/providers/lenovo/physical_infra_manager/physical_switch/operations.rb

```diff
@@ -0,0 +1,34 @@
+#
+# Has the Power Operation methods for Physical Switches
+#
+module ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalSwitch::Operations
+  extend ActiveSupport::Concern
+
+  #
+  # Restarts the physical switch.
+  #   it does the `power_cycle_soft` operation.
+  #
+  def restart
+    change_resource_state(:power_cycle_soft_switch)
+  end
+
+  private
+
+  #
+  # Sends the power operation for a Switch.
+  #
+  # @param [symbol] verb - the operation that must be sent
+  #
+  # @return the LXCA response
+  #
+  def change_resource_state(verb)
```

Comment:
This looks really similar to the method in [PhysicalInfraManager::Operations](https://github.com/ManageIQ/manageiq-providers-lenovo/blob/master/app/models/manageiq/providers/lenovo/physical_infra_manager/operations.rb#L60), I wonder if we can generalize that?  Maybe some future refactoring

Reply:
You're right, but I agree with do this refactoring in a separate PR